src/google/protobuf/compiler/java/map_field.cc

```diff
@@ -335,6 +335,7 @@ void ImmutableMapFieldGenerator::GenerateMembers(io::Printer* printer) const {
       "              $value_default_value$);\n"
       "}\n");
   printer->Print(variables_,
+                 "@SuppressWarnings(\"serial\")\n"
```

Comment:
this may not actually be a false-positive, since I could not find any evidence that any MapField types are actually serializable.

Reply:
I believe that our custom serialization format is implemented in such a way that we dump a bunch of details into the serialization that are not actually used in the deserialization (which is really accomplished through proto deserialization). This is probably fine until we can fix the serialization format to drop things that shouldn't be there.

Reply:
sounds good, thanks for checking